### DISCUSSION TITLE
No such file or directory <nv/target>

### DISCUSSION THREAD

**Discussion by jaisj098** [NVIDIA/cccl#1049] [Thrust]
in `thrust\system\detail\sequential\trivial_copy.h` cant find <nv/target

`
In file included from c:\mingw\include\thrust\system\detail\sequential\copy.inl:23:0,
                 from c:\mingw\include\thrust\system\detail\sequential\copy.h:61,
                 from c:\mingw\include\thrust\system\detail\adl\copy.h:25,
                 from c:\mingw\include\thrust\detail\copy.inl:23,
                 from c:\mingw\include\thrust\detail\copy.h:90,
                 from c:\mingw\include\thrust\detail\allocator\copy_construct_range.inl:23,
                 from c:\mingw\include\thrust\detail\allocator\copy_construct_range.h:45,
                 from c:\mingw\include\thrust\detail\contiguous_storage.inl:23,
                 from c:\mingw\include\thrust\detail\contiguous_storage.h:234,
                 from c:\mingw\include\thrust\detail\temporary_array.h:40,
                 from c:\mingw\include\thrust\system\detail\generic\scan_by_key.inl:27,
                 from c:\mingw\include\thrust\system\detail\generic\scan_by_key.h:142,
                 from c:\mingw\include\thrust\detail\scan.inl:24,
                 from c:\mingw\include\thrust\scan.h:1656,
                 from c:\mingw\include\thrust\system\detail\generic\reduce_by_key.inl:32,
                 from c:\mingw\include\thrust\system\detail\generic\reduce_by_key.h:87,
                 from c:\mingw\include\thrust\detail\reduce.inl:25,
                 from c:\mingw\include\thrust\reduce.h:781,
                 from main.cpp:3:
c:\mingw\include\thrust\system\detail\sequential\trivial_copy.h:27:21: fatal error: nv/target: No such file or directory
 #include <nv/target>
                     ^
compilation terminated.
`

I added thrust and cub to my include path correctly but cant find nv/target

**Comment by alliepiper** [comment]
The `main` branch of Thrust now depends on [libcu++](https://github.com/NVIDIA/libcudacxx) 1.8.0+.

If you're pulling thrust from git, the repo contains a submodule for libcu++ at `<thrust_repo_root>/dependencies/libcudacxx`. This can be synced by running `git submodule update` from the thrust root. Libcu++ 1.8.0 is also shipped in CTK 11.7.

Our CMake build system will automatically locate the libcu++ include paths and configure them appropriately in most cases. For other build system, the include path for libcu++ will need to be specified manually.

Let me know if you run into any issues setting this up and I'll be glad to help.
